### ISSUE TITLE
Docker_registry and nil attributes

### ISSUE THREAD

**Issue by jperville** [bflad/chef-docker#499]
With version 1.0.x of this cookbook I would login (or not) to a docker registry whose URL is provided by the user like this:

```ruby
helper = RegistryUriHelper.new(user_provided_uri)
...
docker_registry helper.registry_host_with_port do
  username helper.registry_username
  password helper.registry_password
  email helper.registry_email
  only_if { helper.registry_username && helper.registry_password }
end
```

After upgrading to version 2.0.0, I get a deprecation warning from chef about setting attribute to nil not overwriting the value of the attribute. Exact error line here: https://github.com/chef-cookbooks/compat_resource/blob/master/files/lib/chef_compat/copied_from_chef/chef/property.rb#L258 . I also triggered a bug in chef_compat, but that's another [story](https://github.com/chef-cookbooks/compat_resource/issues/3). Since I have an `only_if` block in my resource to protect from missing username and password (useful for test registries) I think my use-case is reasonably safe.

I managed to work around the issue on my wrapper cookbook by doing the following (very ugly):

```ruby
  docker_registry helper.registry_host_with_port do
    username helper.registry_username if helper.registry_username # ugly
    password helper.registry_password if helper.registry_password # ugly
    email helper.registry_email if helper.registry_email # ugly!
    only_if { helper.registry_username && helper.registry_password }
  end
```

Would it be possible to make docker_registry accept nil value for all 3 String attributes (username/password/email). I would happily do a PR but I'm short on time this week.

**Comment by someara** [comment]
Seems reasonable. I'll take this one!

**Comment by someara** [comment]
Fixed in cfe0278a3d9e8425bb28a00c8da993c1cfc419ac released as 2.0.3

**Comment by jperville** [comment]
Thanks @somearea you have been faster than me for this PR.
